### ISSUE TITLE
Serious Memory Leak in magick

### ISSUE THREAD

**Issue by 17ssDP** [ImageMagick/ImageMagick#5797]
### ImageMagick version

7.1.0-53

### Operating system

Linux

### Operating system, version and so on

Ubuntu 18.04

### Description

Hi, I found a memory leak vulnerability in magick. It may occupy a great amount of memory when converting an image file.

### Steps to Reproduce

git clone https://github.com/ImageMagick/ImageMagick.git
cd ImageMagick
./configure 
make & make install
/usr/local/bin/magick convert poc /dev/null

### Images

[poc.zip](https://github.com/ImageMagick/ImageMagick/files/10075563/poc.zip)

**Comment by dlemstra** [comment]
Can you reproduce this issue in our GitHub Codespace?

**Comment by 17ssDP** [comment]
Yes, I can reproduce it in the GitHub Codespace. It consumes all the space and leads to the following error.

 /workspaces/ImageMagick/utilities (main ✗) $ ./magick convert ../poc /dev/null 
convert: unable to write pixel cache '/tmp/magick-MOOu8P_QVu0yQrz2em7c3rj_QTVdjjNM': No space left on device @ error/cache.c/WritePixelCachePixels/5852.
convert: no images defined `/dev/null' @ error/convert.c/ConvertImageCommand/3342.
